lib/web/FetchCompileWasmMainTemplatePlugin.js

```diff
@@ -26,6 +27,88 @@ class FetchCompileWasmMainTemplatePlugin {
 			(source, chunk, hash) => {
 				const webassemblyModuleFilename =
 					mainTemplate.outputOptions.webassemblyModuleFilename;
+
+				// Get all wasm modules
+				function getAllWasmModules() {
+					const wasmModules = chunk.getAllAsyncChunks();
+					const array = [];
+					for (const chunk of wasmModules) {
+						for (const m of chunk.modulesIterable) {
+							if (m.type.startsWith("webassembly")) {
+								array.push(m);
+							}
+						}
+					}
+
+					return array;
+				}
+
+				function generateImportObject(module) {
+					const depsByRequest = new Map();
+					for (const dep of module.dependencies) {
+						if (dep instanceof WebAssemblyImportDependency) {
+							// Ignore global they will be handled later
+							if (dep.description.type === "GlobalType") {
+								continue;
+							}
+
+							const request = dep.request;
+							let array = depsByRequest.get(request);
+							if (!array) {
+								depsByRequest.set(request, (array = []));
+							}
+							const exportName = dep.name;
+							const usedName = dep.module && dep.module.isUsed(exportName);
+							array.push({
+								exportName,
+								usedName,
+								module: dep.module,
+								description: dep.description
+							});
+						}
+					}
+					const importsCode = [];
+					for (const pair of depsByRequest) {
+						const properties = [];
+						for (const data of pair[1]) {
+							let params = "";
+							let result = "void 0";
+
+							if (data.description.type === "FuncImportDescr") {
+								params = data.description.params.map(
+									(param, k) => "p" + k + param.valtype
+								);
+
+								result = `__webpack_require__(${JSON.stringify(
+									data.module.id
+								)})[${JSON.stringify(data.usedName)}](${params})`;
+							}
+
+							if (data.description.type === "GlobalType") {
```

Comment:
Is this used? I guess all globals are filtered by the first loop.

Reply:
Yes, I removed it